app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -538,12 +560,114 @@ fun MessageList(
+                                    assetSizeInBytes = messageContent.assetSizeInBytes,
+                                    assetUploadStatus = messageContent.uploadStatus,
+                                    assetDownloadStatus = messageContent.downloadStatus,
+                                    onAssetClick = assetClickable
+                                )
+                            }
+
+                            is UIMessageContent.RestrictedAsset -> {
+                                MessageRestrictedAsset(
+                                    assetName = messageContent.assetName,
+                                    assetSizeInBytes = messageContent.assetSizeInBytes,
+                                    mimeType = messageContent.mimeType
+                                )
+                            }
+
+                            is UIMessageContent.AudioAssetMessage -> {
+                                // in case the user did not play the audio message yet, we don't have the audio state
+                                // that is coming from ConversationMessageAudioPlayer, in that case we refer to
+                                // the default values for the audio message, which is PAUSED state and 0 as current position
+                                // we treat totalTimeInMs from backend as the source of truth
+                                // in order to be sure that all clients are in sync regarding the audio message duration
+                                // however if it returns 0 we fallback to the duration from the audio player
+                                val audioMessageState: AudioState =
+                                    audioMessagesState[message.messageHeader.messageId] ?: AudioState.DEFAULT
+
+                                val totalTimeInMs = if (messageContent.audioMessageDurationInMs.toInt() == 0) {
+                                    audioMessageState.totalTimeInMs
+                                } else {
+                                    messageContent.audioMessageDurationInMs.toInt()
+                                }
+
+                                AudioMessage(
+                                    audioMediaPlayingState = audioMessageState.audioMediaPlayingState,
+                                    totalTimeInMs = totalTimeInMs,
+                                    currentPositionInMs = audioMessageState.currentPositionInMs,
+                                    onPlayButtonClick = { onAudioClick(message.messageHeader.messageId) },
+                                    onSliderPositionChange = { position ->
+                                        onChangeAudioPosition(message.messageHeader.messageId, position.toInt())
+                                    },
+                                    onAudioMessageLongClick = { onShowContextMenu(message) }
+                                )
+                            }
```

Comment:
I decided to add messageContent argument to MessageItem, because it was a hell of passing all those callback that in the end are most of the time not consumed. I am talking here about the case where the content is of MessageText, but we will do not know it yet, so we are passing all those callbacks for different type content, this could be further improved but I did not want to make a PR even bigger